From 4a3cebec79d2f55859a9b7ff416afd09406f7218 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Thu, 2 Dec 2021 10:05:12 +1100
Subject: [PATCH 2/5] do not build protobuf

---
 third_party/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index d00ecba..b355ccc 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -1,4 +1,4 @@
-include_directories(absl/strings darts_clone esaxx protobuf-lite)
+include_directories(absl/strings darts_clone esaxx)
 
 
 
-- 
2.32.0.windows.2

