From e938934e6582cdb4e968aad70a38d09e2be667a3 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 6 Dec 2021 21:18:54 +1100
Subject: [PATCH 3/9] consistently use absolute paths for CMAKE_INSTALL_*DIR

---
 CMakeLists.txt | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 827d114..f4f7548 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -40,11 +40,8 @@ endif()
 
 if (UNIX)
   include(GNUInstallDirs)
-  set(prefix ${CMAKE_INSTALL_PREFIX})
-  set(exec_prefix "\${prefix}")
-  set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
-  set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
-else()
+endif()
+if (TRUE)
   set(prefix ${CMAKE_INSTALL_PREFIX})
   set(exec_prefix "\${prefix}")
   set(libdir "\${exec_prefix}/lib")
@@ -83,15 +80,15 @@ if (APPLE)
 endif()
 
 if (NOT DEFINED CMAKE_INSTALL_BINDIR)
-  set(CMAKE_INSTALL_BINDIR bin)
+  set(CMAKE_INSTALL_BINDIR "${CMAKE_INSTALL_PREFIX}/bin")
 endif()
 
 if (NOT DEFINED CMAKE_INSTALL_LIBDIR)
-  set(CMAKE_INSTALL_LIBDIR lib)
+  set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_PREFIX}/lib")
 endif()
 
 if (NOT DEFINED CMAKE_INSTALL_INCDIR)
-  set(CMAKE_INSTALL_INCDIR include)
+  set(CMAKE_INSTALL_INCDIR "${CMAKE_INSTALL_PREFIX}/include")
 endif()
 
 configure_file("${PROJECT_SOURCE_DIR}/config.h.in" "config.h")
-- 
2.38.1.windows.1

