From bb0bda21c70a52c388353876a64560b5f6243c7b Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 6 Dec 2021 21:18:54 +1100
Subject: [PATCH 3/8] consistently use absolute paths for CMAKE_INSTALL_*DIR

---
 CMakeLists.txt | 13 +++++--------
 1 file changed, 5 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e40f5c..434530a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,11 +51,8 @@ endif()
 
 if (UNIX)
   include(GNUInstallDirs)
-  set(prefix ${CMAKE_INSTALL_PREFIX})
-  set(exec_prefix "\${prefix}")
-  set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
-  set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
-else()
+endif()
+if (TRUE)
   set(prefix ${CMAKE_INSTALL_PREFIX})
   set(exec_prefix "\${prefix}")
   set(libdir "\${exec_prefix}/lib")
@@ -96,15 +93,15 @@ if (APPLE)
 endif()
 
 if (NOT DEFINED CMAKE_INSTALL_BINDIR)
-  set(CMAKE_INSTALL_BINDIR bin)
+  set(CMAKE_INSTALL_BINDIR "${CMAKE_INSTALL_PREFIX}/bin")
 endif()
 
 if (NOT DEFINED CMAKE_INSTALL_LIBDIR)
-  set(CMAKE_INSTALL_LIBDIR lib)
+  set(CMAKE_INSTALL_LIBDIR "${CMAKE_INSTALL_PREFIX}/lib")
 endif()
 
 if (NOT DEFINED CMAKE_INSTALL_INCDIR)
-  set(CMAKE_INSTALL_INCDIR include)
+  set(CMAKE_INSTALL_INCDIR "${CMAKE_INSTALL_PREFIX}/include")
 endif()
 
 # SPDX-License-Identifier: (MIT OR CC0-1.0)
