From 9580d279f6ed7c6eb865d384dbc9945b4b73735e Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 6 Dec 2021 21:18:54 +1100
Subject: [PATCH 4/8] fix libdir

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 62c3b09..b097e94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -42,8 +42,8 @@ if (UNIX)
   include(GNUInstallDirs)
   set(prefix ${CMAKE_INSTALL_PREFIX})
   set(exec_prefix "\${prefix}")
-  set(libdir "\${exec_prefix}/${CMAKE_INSTALL_LIBDIR}")
-  set(includedir "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+  set(libdir ${CMAKE_INSTALL_LIBDIR})
+  set(includedir ${CMAKE_INSTALL_INCLUDEDIR})
 else()
   set(prefix ${CMAKE_INSTALL_PREFIX})
   set(exec_prefix "\${prefix}")
-- 
2.32.0.windows.2

