From a5e6c739438f4cb97c04272b90f5bfcf33529aca Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 13 Feb 2022 15:45:51 +1100
Subject: [PATCH 6/9] patch generated protobuf sources

---
 src/CMakeLists.txt | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 418b963..c7b66b7 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -78,8 +78,37 @@ else()
   include_directories(${Protobuf_INCLUDE_DIRS})
   protobuf_generate_cpp(SPM_PROTO_SRCS SPM_PROTO_HDRS sentencepiece.proto)
   protobuf_generate_cpp(SPM_MODEL_PROTO_SRCS SPM_MODEL_PROTO_HDRS sentencepiece_model.proto)
+  set_source_files_properties(sentencepiece.vcxproj
+                              sentencepiece_model.pb.h
+                              sentencepiece_model.pb.cc
+                              PROPERTIES GENERATED TRUE)
   set(PROTOBUF_LITE_SRCS "")
   include_directories(${PROTOBUF_INCLUDE_DIR})
+  if (MSVC)
+    # apply patch so that windows can compile; need to make sure this runs
+    # at "--build"-time (i.e. add_custom_command instead of execute_process),
+    # as that's when protobuf_generate_cpp actually produces the sources;
+    # use DEPENDS to ensure we run afer protobuf
+    add_custom_command(OUTPUT sentencepiece_model_patched.pb.cc
+                       DEPENDS sentencepiece.vcxproj
+                               sentencepiece_model.pb.h sentencepiece_model.pb.cc
+                       COMMAND echo "conda-forge build debug: begin sentencepiece_model.pb.h"
+                               && type sentencepiece_model.pb.h
+                               && echo "conda-forge build debug: end sentencepiece_model.pb.h"
+                               && echo "conda-forge build debug: begin sentencepiece_model.pb.cc"
+                               && type sentencepiece_model.pb.cc
+                               && echo "conda-forge build debug: end sentencepiece_model.pb.cc"
+                               && dir .
+                               && dir CMakeFiles
+                               && dir CMakeFiles\\sentencepiece.dir
+                               && echo "conda-forge build debug: begin sentencepiece.vcxproj"
+                               && type sentencepiece.vcxproj
+                               && echo "conda-forge build debug: end sentencepiece.vcxproj"
+                               && patch -Np1 -i "$ENV{RECIPE_DIR}\\patch_generated_proto.patch" --binary
+                               && move sentencepiece_model.pb.cc sentencepiece_model_patched.pb.cc)
+    # update name so that correct build order is maintained; don't change header name
+    set(SPM_MODEL_PROTO_SRCS "sentencepiece_model_patched.pb.cc")
+  endif()
 endif()
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
-- 
2.37.0.windows.1

