From d402490f8eb8c50ea357a674a77d343c54a37921 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Sun, 13 Feb 2022 15:45:51 +1100
Subject: [PATCH 7/7] patch generated protobuf sources

---
 src/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c654f29..1b67265 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -76,6 +76,10 @@ else()
   protobuf_generate_cpp(SPM_MODEL_PROTO_SRCS SPM_MODEL_PROTO_HDRS sentencepiece_model.proto)
   set(PROTOBUF_LITE_SRCS "")
   include_directories(${PROTOBUF_INCLUDE_DIR})
+  if (MSVC)
+    # apply patch so that windows can compile
+    execute_process(COMMAND patch -Np1 -i "$ENV{RECIPE_DIR}\\patch_generated_proto.patch" --binary)
+  endif()
 endif()
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
-- 
2.32.0.windows.2

