From a9d4cda89f8eb9a13e37a8e2a74908d615c23c29 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 21 Feb 2022 16:07:24 +1100
Subject: [PATCH 9/9] debug linker errors

---
 src/CMakeLists.txt | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0ed702e..818dfa9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -256,6 +256,23 @@ if (SPM_ENABLE_SHARED)
   if (MSVC)
     set_target_properties(sentencepiece PROPERTIES IMPORT_SUFFIX "_import.lib")
     set_target_properties(sentencepiece_train PROPERTIES IMPORT_SUFFIX "_import.lib")
+    target_link_options(sentencepiece PUBLIC "/verbose" "/NODEFAULTLIB:libcmt.lib libcpmt.lib")
+    target_link_options(sentencepiece_train PUBLIC "/verbose" "/NODEFAULTLIB:libcmt.lib libcpmt.lib")
+    set(MSVC_C_CXX_FLAGS
+      CMAKE_C_FLAGS_DEBUG
+      CMAKE_C_FLAGS_MINSIZEREL
+      CMAKE_C_FLAGS_RELEASE
+      CMAKE_C_FLAGS_RELWITHDEBINFO
+      CMAKE_CXX_FLAGS_DEBUG
+      CMAKE_CXX_FLAGS_MINSIZEREL
+      CMAKE_CXX_FLAGS_RELEASE
+      CMAKE_CXX_FLAGS_RELWITHDEBINFO
+    )
+    foreach(flag ${MSVC_C_CXX_FLAGS})
+      if(${flag} MATCHES "/MT")
+        string(REGEX REPLACE "/MT" "/MD" ${flag} "${${flag}}")
+      endif()
+    endforeach()
   elseif (MINGW)
     set_target_properties(sentencepiece PROPERTIES IMPORT_SUFFIX ".dll.a")
     set_target_properties(sentencepiece_train PROPERTIES IMPORT_SUFFIX ".dll.a")
-- 
2.37.0.windows.1

