From 22064729a2dc0dd519600c28929812bc3dd7f0a8 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 25 Nov 2024 20:29:08 +1100
Subject: [PATCH 10/12] avoid spuriously inclusing vendored abseil

---
 src/CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 91ee1e8..6ffff3e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -107,7 +107,6 @@ elseif (SPM_PROTOBUF_PROVIDER STREQUAL "package")
 endif()
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/../third_party)
 
 if (MSVC)
   add_definitions("/D_USE_INTERNAL_STRING_VIEW")
