From 4f262becfff43be0c8847cd3f9d3c2d4a1121c14 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 25 Nov 2024 20:29:08 +1100
Subject: [PATCH 11/13] fix minloglevel situation; avoid _USE_EXTERNAL_ABSL

---
 src/CMakeLists.txt | 5 -----
 src/init.h         | 5 ++---
 2 files changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 6ffff3e..16621e9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -20,11 +20,6 @@ if (SPM_ABSL_PROVIDER STREQUAL "module" OR SPM_ABSL_PROVIDER STREQUAL "package")
   list(APPEND SPM_LIBS absl::flags_parse)
   list(APPEND SPM_LIBS absl::log)
   list(APPEND SPM_LIBS absl::check)
-  if (MSVC)
-   add_definitions("/D_USE_EXTERNAL_ABSL")
-  else()
-   add_definitions("-D_USE_EXTERNAL_ABSL")
-  endif()
 elseif (SPM_ABSL_PROVIDER STREQUAL "internal")
   set(ABSL_FLAGS_SRCS ${CMAKE_CURRENT_SOURCE_DIR}/../third_party/absl/flags/flag.cc)
 endif()
diff --git a/src/init.h b/src/init.h
index 5a69100..be2e713 100644
--- a/src/init.h
+++ b/src/init.h
@@ -19,6 +19,7 @@
 #include "third_party/absl/flags/flag.h"
 #include "third_party/absl/flags/parse.h"
 #include "third_party/absl/flags/usage.h"
+#include "absl/log/globals.h"
 
 #ifdef _USE_EXTERNAL_PROTOBUF
 #include "google/protobuf/message_lite.h"
@@ -26,8 +27,6 @@
 #include "third_party/protobuf-lite/google/protobuf/message_lite.h"
 #endif
 
-ABSL_DECLARE_FLAG(int32_t, minloglevel);
-
 namespace sentencepiece {
 inline void ParseCommandLineFlags(const char *usage, int *argc, char ***argv,
                                   bool remove_arg = true) {
@@ -41,7 +40,7 @@ inline void ParseCommandLineFlags(const char *usage, int *argc, char ***argv,
     *argc = static_cast<int>(unused_args.size());
   }
 
-  logging::SetMinLogLevel(absl::GetFlag(FLAGS_minloglevel));
+  logging::SetMinLogLevel((int)absl::MinLogLevel());
 }
 
 inline void ShutdownLibrary() {
