From ccd4b762c899dc2ced8bc9cdf095b7890ed07e66 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Mon, 25 Nov 2024 20:59:56 +1100
Subject: [PATCH 13/13] adapt to names for import_library

---
 python/setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/setup.py b/python/setup.py
index 1316df3..13b33d6 100755
--- a/python/setup.py
+++ b/python/setup.py
@@ -131,7 +131,7 @@ if os.name == 'nt':
       # protobuf actually has the lib-prefix in the name also on windows;
       # since libsentencepiece is static on windows, we also need _its_
       # host dependencies for the link interface, i.e. also abseil
-      for x in ["sentencepiece", "sentencepiece_train", "libprotobuf"] + absl_libs
+      for x in ["sentencepiece_import", "sentencepiece_train_import", "libprotobuf"] + absl_libs
     ]
   else:
     # build library locally with cmake and vc++.
